src/library/scala/collection/immutable/List.scala

```diff
@@ -108,7 +108,20 @@ sealed abstract class List[+A]
   def ::: [B >: A](prefix: List[B]): List[B] =
     if (isEmpty) prefix
     else if (prefix.isEmpty) this
-    else (new ListBuffer[B] ++= prefix).prependToList(this)
+    else {
+      val result = new ::[B](prefix.head, Nil)
+      var curr = result
+      var that = prefix.tail
+      while (that.nonEmpty) {
+        val temp = new ::[B](that.head, Nil)
```

Comment:
I don't know how much it'd help, but you could change this to `new ::[B](that.head, null)` to avoid loading `Nil`.

Reply:
Thanks for the tip, I'll try it out.